dGame/dComponents/Component.h

```diff
@@ -43,10 +44,17 @@ class Component
      */
     virtual void LoadFromXml(tinyxml2::XMLDocument* doc);
 
+    using Handler = void(Component::*)(GameMessage*);
```

Comment:
We use std::function in other places for callbacks, it'd probably be wise to use it here too.

Reply:
`std::function` seems just like a less memory efficient way of doing the same thing.

Reply:
> `std::function` seems just like a less memory efficient way of doing the same thing.

The same could be said about game message objects.